src/GHCMod.hs

```diff
@@ -333,7 +342,14 @@ main = handler $ do
 
 progMain :: (Options,[String]) -> IO ()
 progMain (globalOptions,cmdArgs) = do
-    (res,_) <- runGhcModT globalOptions $ ghcCommands cmdArgs
+    let
```

Comment:
How about just passing the FileMapping down to a new variant of the run* functions then?